builtin/logical/pki/cert_template.proto

```diff
@@ -0,0 +1,211 @@
+syntax = "proto3";
+
+package openbao.pki;
+
+option go_package = "github.com/openbao/openbao/builtin/logical/pki";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/any.proto";
+
+message PKIX {
+    message Name {
+        repeated string Country = 1;
+        repeated string Organization = 2;
+        repeated string OrganizationUnit = 3;
+        repeated string Locality = 4;
+        repeated string Province = 5;
+        repeated string StreetAddress = 6;
+        repeated string PostalCode = 7;
+        string SerialNumber = 8;
+        string CommonName = 9;
+    }
+    message Extension {
+        repeated int64 Id  = 1;
+        bool Critical = 2;
+        bytes Value = 3;
+    }
+}
+
+message ObjectIdentifier {
+  // 1.2.840.113549 → [1, 2, 840, 113549]
+  repeated uint32 arc = 1;
+}
+
+message IPMask {
+    bytes IPMask = 1;
+}
+
+message net {
+    message IP {
+        bytes IP = 1;
+    }
+    message IPNet {
+        net.IP IP = 1;          // network number
+	    IPMask Mask = 2;
+    }
+}
+
+message Userinfo {
+	string username      = 1;   // RFC‑2396 allows empty.
+    string password      = 2;   // Empty when not supplied.
+    bool   password_set  = 3;   // If true, then password field is meaningful.
+}
+
+message url {
+    message URL {
+        string Scheme = 1;
+        string Opaque = 2;
+        Userinfo User = 3;
+        string Host = 4;
+        string Path = 5;
+        string RawPath = 6;
+        bool OmitHost = 7;
+        bool ForceQuery = 8;
+        string RawQuery = 9;
+        string Fragment = 10;
+        string RawFragment = 11;
+    }    
+}
+
+// An OID represents an ASN.1 OBJECT IDENTIFIER.
+message OID {
+    repeated uint64 arc = 1;     // 1.2.840.113549 → [1, 2, 840, 113549]
+}
+
+message PolicyMapping {	
+	OID IssuerDomainPolicy = 1;	
+	OID SubjectDomainPolicy = 2;
+}
+
+enum PublicKeyAlgorithm {
+    UnknownPublicKeyAlgorithm = 0;
+    RSA = 1;
+    DSA = 2;
+    ECDSA = 3;
+    Ed25519 = 4;
+}
+
+enum SignatureAlgorithm {
+    UnknownSignatureAlgorithm = 0;
+	MD2WithRSA  = 1;            // Unsupported.
+	MD5WithRSA = 2;             // Only supported for signing, not verification.
+	SHA1WithRSA = 3;            // Only supported for signing, and verification of CRLs, CSRs, and OCSP responses.
+	SHA256WithRSA = 4;
+	SHA384WithRSA = 5;
+	SHA512WithRSA = 6;
+	DSAWithSHA1 = 7;            // Unsupported.
+	DSAWithSHA256 = 8;          // Unsupported.
+	ECDSAWithSHA1 = 9;          // Only supported for signing, and verification of CRLs, CSRs, and OCSP responses.
+	ECDSAWithSHA256 = 10;
+	ECDSAWithSHA384 = 11;
+	ECDSAWithSHA512 = 12;
+	SHA256WithRSAPSS = 13;
+	SHA384WithRSAPSS = 14;
+	SHA512WithRSAPSS = 15;
+	PureEd25519 = 16;
+}
```

Comment:
I think we can drop this field; it will be determined by PKI internally (+/- `use_pss` as a flag on the role) and won't really be a CEL-author decision. Half of the value is determined by the issuer key type anyways.